pkg/ccl/sqlproxyccl/server.go

```diff
@@ -349,6 +395,52 @@ func (s *Server) AwaitNoConnections(ctx context.Context) <-chan struct{} {
 	return c
 }
 
+// shouldLogError returns true if an error should be logged, or false otherwise.
+// The goal is to only throttle high-frequency errors for every (IP, tenantID)
+// pair.
+func (s *Server) shouldLogError(err error, conn net.Conn, reqTags map[string]interface{}) bool {
```

Comment:
nit: not a strong opinion so feel free to ignore, but I'm in favor of having the second argument be something like `remoteAddr string` instead of `conn net.Conn`. It looks a little funny to me at first glance that this func has to take in a `net.Conn` object.

Reply:
I left it as-is. I didn't want to do the parsing of the IP outside of `shouldLogError`. An alternative design is to take in an interface that only implements `RemoteAddr()`.